src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+            New WellKnownMember() {WellKnownMember.System_ValueTuple_T7__Item1, WellKnownMember.System_ValueTuple_T7__Item2, WellKnownMember.System_ValueTuple_T7__Item3, WellKnownMember.System_ValueTuple_T7__Item4, WellKnownMember.System_ValueTuple_T7__Item5, WellKnownMember.System_ValueTuple_T7__Item6, WellKnownMember.System_ValueTuple_T7__Item7},
+            New WellKnownMember() {WellKnownMember.System_ValueTuple_TRest__Item1, WellKnownMember.System_ValueTuple_TRest__Item2, WellKnownMember.System_ValueTuple_TRest__Item3, WellKnownMember.System_ValueTuple_TRest__Item4, WellKnownMember.System_ValueTuple_TRest__Item5, WellKnownMember.System_ValueTuple_TRest__Item6, WellKnownMember.System_ValueTuple_TRest__Item7, WellKnownMember.System_ValueTuple_TRest__Rest}}
+
+        Public Overrides ReadOnly Property IsTupleType() As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TupleUnderlyingType() As NamedTypeSymbol
+            Get
+                Return Me._underlyingType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TupleElementTypes() As ImmutableArray(Of TypeSymbol)
+            Get
+                Return Me._elementTypes
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property TupleElementNames() As ImmutableArray(Of String)
+            Get
+                Return Me._elementNames
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsReferenceType() As Boolean
+            Get
+                Return Not Me._underlyingType.IsErrorType() AndAlso Me._underlyingType.IsReferenceType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsValueType() As Boolean
+            Get
+                Return Me._underlyingType.IsValueType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsImplicitlyDeclared() As Boolean
+            Get
+                Return False
+            End Get
+        End Property
+
+        Public ReadOnly Property TupleElementFields() As ImmutableArray(Of FieldSymbol)
+            Get
+                Dim isDefault As Boolean = Me._lazyFields.IsDefault
+                If isDefault Then
+                    ImmutableInterlocked.InterlockedInitialize(Of FieldSymbol)(Me._lazyFields, Me.CollectTupleElementFields())
+                End If
+                Return Me._lazyFields
+            End Get
+        End Property
+
+        Friend ReadOnly Property UnderlyingDefinitionToMemberMap() As SmallDictionary(Of Symbol, Symbol)
+            Get
+                If Me._lazyUnderlyingDefinitionToMemberMap Is Nothing Then
+                    Me._lazyUnderlyingDefinitionToMemberMap = Me.ComputeDefinitionToMemberMap()
```

Comment:
Use `Interlocked.CompareExchange` to avoid having two callers see different maps (and therefore avoid two instances of the same members).